main.go

```diff
@@ -112,22 +125,45 @@ func main() {
 	flag.Parse()
 
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+	config := ctrl.GetConfigOrDie()
+
+	if flagVar.initOnly {
```

Comment:
We should update the ADR to also document why we now use have an init only mode and an init container.